lib/database_transform/schema_table.rb

```diff
@@ -131,8 +134,14 @@ def time_block(before, after, &proc)
   def transform!(schema)
     # For each item in the old model
     default_scope = @default_scope || @source.method(:all)
-    @source.instance_exec(&default_scope).each do |record|
-      transform_record!(schema, record)
+    collection = @source.instance_exec(&default_scope)
+
+    if collection.respond_to?(:find_in_batches)
```

Comment:
alternatively you can write another method to simulate find_in_batches, transform_in_batches! maybe?